### PR TITLE
Fix network stats to use low-level values

### PR DESCRIPTION
**Pull request by Atlante45**
https://highfidelity.fogbugz.com/f/cases/20163/Avatar-mixer-stats-page-should-include-identity-traits-bytes

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15975/HighFidelity-Beta-PR14591-ee2d33e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15975/HighFidelity-Beta-Interface-PR14591-ee2d33e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15975/HighFidelity-Beta-PR14591-ee2d33e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15975/HighFidelity-Beta-Interface-PR14591-ee2d33e.exe

**Comment by howard-stearns** [comment]
Does this cover https://highfidelity.fogbugz.com/f/cases/20163/Avatar-mixer-stats-page-should-include-identity-traits-bytes?

File: libraries/networking/src/Node.h [hunk 1]
```diff
@@ -115,6 +126,8 @@ class Node : public NetworkPeer {
     IgnoredNodeIDs _ignoredNodeIDs;
     mutable QReadWriteLock _ignoredNodeIDSetLock;
     std::vector<QString> _replicatedUsernames { };
+
+    Stats _stats;
 };
```
Comment (SimonWalton-HiFi) [inline libraries/networking/src/Node.h]: Are we keeping ConnectionStats in both the Node and Connection classes now?

File: libraries/networking/src/udt/Connection.h [hunk 2]
```diff
@@ -73,14 +73,18 @@ class Connection : public QObject {
     void setMaxBandwidth(int maxBandwidth);
 
     void sendHandshakeRequest();
+    
+    void recordSentUnreliablePackets(int wireSize, int payloadSize);
+    void recordRecievedUnreliablePackets(int wireSize, int payloadSize);
```
Comment (SimonWalton-HiFi) [inline libraries/networking/src/udt/Connection.h]: **Received**

File: libraries/networking/src/LimitedNodeList.cpp [hunk 3]
```diff
@@ -887,10 +877,56 @@ void LimitedNodeList::removeSilentNodes() {
     }
 }
 
+void LimitedNodeList::sampleConnectionStats() {
+    uint32_t packetsIn { 0 };
```
Comment (SimonWalton-HiFi) [inline libraries/networking/src/LimitedNodeList.cpp]: This routine, and getInboundPPS() etc, just refer to reliable traffic - from the Connection class - right? Maybe a comment to this effect would help.

File: libraries/networking/src/udt/Socket.cpp [hunk 4]
```diff
@@ -392,9 +398,10 @@ void Socket::readPendingDatagrams() {
 
             // call our verification operator to see if this packet is verified
             if (!_packetFilterOperator || _packetFilterOperator(*packet)) {
+                auto connection = findOrCreateConnection(senderSockAddr, true);
+
```
Comment (SimonWalton-HiFi) [inline libraries/networking/src/udt/Socket.cpp]: So is the new scheme to create a Connection class even if we never use reliable service to a node, and use that to track stats? Are there any ramifications of always creating a Connectopn class?

File: libraries/networking/src/Node.h (hunk 1 above)
Comment (Atlante45) [reply libraries/networking/src/Node.h]: Yes, the Connection is where we store the stats while they are being gathered.
And the Node is where we store the sampled stats of the full, past second.

File: libraries/networking/src/udt/Connection.h (hunk 2 above)
Comment (Atlante45) [reply libraries/networking/src/udt/Connection.h]: =(

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16214/HighFidelity-Beta-PR14591-a0ede77.apk

File: libraries/networking/src/udt/Socket.cpp (hunk 4 above)
Comment (Atlante45) [reply libraries/networking/src/udt/Socket.cpp]: As long as the send queue is not started, the connection won't do anything except store some data.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16214/HighFidelity-Beta-PR14591-a0ede77.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16214/HighFidelity-Beta-Interface-PR14591-a0ede77.dmg

File: libraries/networking/src/LimitedNodeList.cpp (hunk 3 above)
Comment (Atlante45) [reply libraries/networking/src/LimitedNodeList.cpp]: I add up both, they refer to the overall traffic.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16214/HighFidelity-Beta-PR14591-a0ede77.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16214/HighFidelity-Beta-Interface-PR14591-a0ede77.exe

**Pull request merged**